tools/msys2_fetch_and_build_all_tx16s.sh

```diff
@@ -15,81 +150,111 @@ do
+  cd ${SOURCE_DIR}
+else
+  echo "=== Step $((STEP++)): updating EdgeTX source tree (${BRANCH_NAME} branch) from GitHub ==="
+  cd ${SOURCE_DIR}
+  check_command $? "cd ${SOURCE_DIR}"
+  git checkout ${BRANCH_NAME}
+  check_command $? "git checkout ${BRANCH_NAME}"
+  git pull
+  check_command $? "git pull"
+  git submodule update --init --recursive
+  check_command $? "git submodule update --init --recursive"
+fi
 if [[ $PAUSEAFTEREACHLINE == "true" ]]; then
   echo "Step finished. Please check the output above and press Enter to continue or Ctrl+C to stop."
   read
 fi
 
 echo "=== Step $((STEP++)): Creating build output directory ==="
-cd edgetx_main && mkdir build-output && cd build-output
+[[ -d ${BUILD_OUTPUT_DIR} ]] && rm -rf ${BUILD_OUTPUT_DIR}
+mkdir -p ${BUILD_OUTPUT_DIR} && cd ${BUILD_OUTPUT_DIR}
+check_command $? "mkdir -p ${BUILD_OUTPUT_DIR} && cd ${BUILD_OUTPUT_DIR}"
 if [[ $PAUSEAFTEREACHLINE == "true" ]]; then
   echo "Step finished. Please check the output above and press Enter to continue or Ctrl+C to stop."
   read
 fi
 
-echo "=== Step $((STEP++)): Running CMake for TX16S as an example ==="
-cmake -G "MSYS Makefiles" -Wno-dev -DCMAKE_PREFIX_PATH=$HOME/5.12.9/mingw73_64 -DSDL_LIBRARY_PATH=/mingw64/bin/ -DPCB=X10 -DPCBREV=TX16S -DDEFAULT_MODE=2 -DGVARS=YES -DPPM_UNIT=US -DHELI=NO -DLUA=YES -DCMAKE_BUILD_TYPE=Release ../
+echo "=== Step $((STEP++)): Running CMake for ${RADIO_TYPE} as an example ==="
+cmake -G "MSYS Makefiles" -Wno-dev -DCMAKE_PREFIX_PATH=$HOME/5.12.9/mingw73_64 -DSDL_LIBRARY_PATH=/mingw64/bin/ ${BUILD_OPTIONS} -DDEFAULT_MODE=2 -DGVARS=YES -DPPM_UNIT=US -DHELI=NO -DLUA=YES -DCMAKE_BUILD_TYPE=Release ../
+check_command $? "cmake -G MSYS Makefiles -Wno-dev -DCMAKE_PREFIX_PATH=$HOME/5.12.9/mingw73_64 -DSDL_LIBRARY_PATH=/mingw64/bin/ ${BUILD_OPTIONS} -DDEFAULT_MODE=2 -DGVARS=YES -DPPM_UNIT=US -DHELI=NO -DLUA=YES -DCMAKE_BUILD_TYPE=Release ../"
 if [[ $PAUSEAFTEREACHLINE == "true" ]]; then
   echo "Step finished. Please check the output above and press Enter to continue or Ctrl+C to stop."
   read
 fi
 
 echo "=== Step $((STEP++)): Running Make configure ==="
 make configure
+check_command $? "make configure"
 if [[ $PAUSEAFTEREACHLINE == "true" ]]; then
   echo "Step finished. Please check the output above and press Enter to continue or Ctrl+C to stop."
   read
 fi
 
 echo "=== Step $((STEP++)): Building firmware binary ==="
 make -C arm-none-eabi -j`nproc` firmware
+check_command $? "make -C arm-none-eabi -j`nproc` firmware"
 if [[ $PAUSEAFTEREACHLINE == "true" ]]; then
   echo "Step finished. Please check the output above and press Enter to continue or Ctrl+C to stop."
   read
 fi
 
 echo "=== Step $((STEP++)): Renaming firmware binary ==="
-mv arm-none-eabi/firmware.bin arm-none-eabi/fw_edgetx_main_tx16s_lua-ppmus-mode2_release.bin
+mv arm-none-eabi/firmware.bin arm-none-eabi/fw_${SOURCE_DIR}_${RADIO_TYPE}_lua-ppmus-mode2_release.bin
```

Comment:
Here is a bug, please run the script to test and see this: folder `arm-none-eabi/fw_${SOURCE_DIR}` does not exist (SOURCE_DIR is evaluated to `${HOME}/edgetx/edgetx_${BRANCH_NAME}`). Try instead:
`mv arm-none-eabi/firmware.bin arm-none-eabi/fw_${RADIO_TYPE}_lua-ppmus-mode2_release.bin`

The same applies to the next line as well.